xfrd.c

```diff
@@ -2739,6 +2741,70 @@ xfrd_handle_taskresult(xfrd_state_type* xfrd, struct task_list_d* task)
 		xfrd_process_zonestat_inc_task(xfrd, task);
 		break;
 #endif
+	case task_add_catzone: {
+		char* zname = (char*)task->zname;
+		char* pattern = zname + strlen(zname) + 1;
+		char* catname = pattern + strlen(pattern) + 1;
+		char* member_id = catname + strlen(catname) + 1;
+
+		struct zone_options* zopt = NULL;
+		pattern_options_type* patopt;
+
+		log_msg(LOG_INFO, "now adding catzone %s %s %s %s", zname, pattern, catname, member_id);
+
+		patopt = pattern_options_find(xfrd->nsd->options, pattern);
+		if (!patopt) {
+			patopt = pattern_options_create(xfrd->region);
+			patopt->pname = pattern;
```

Comment:
The variable pattern is part of the task, that is deleted after the function ends. But pattern is added to the options. It should have contents that is allocated.